client/util.go

```diff
@@ -0,0 +1,116 @@
+package client
+
+import (
+	"bytes"
+	"crypto/rand"
+	"encoding/base64"
+	"encoding/binary"
+	"errors"
+	"io"
+
+	sserver1 "github.com/keybase/search/protocol/sserver"
+	"golang.org/x/crypto/nacl/secretbox"
+)
+
+// The length of the overhead added to padding.
+const padPrefixLength = 4
+
+// pathnameToDocID encrypts a `pathname` to a document ID using `key`.
+func pathnameToDocID(pathname string, key *[32]byte) (sserver1.DocumentID, error) {
```

Comment:
I know `Seal` takes a pointer, but I don't think there's any reason to take one in these methods (since it'll never be nil).